synapse/rest/client/room.py

```diff
@@ -560,6 +591,7 @@ def __init__(self, hs: "HomeServer"):
         self.auth = hs.get_auth()
         self.store = hs.get_datastores().main
 
+    @messsages_response_timer.time()
```

Comment:
I don't think this is working how we expect. All of the buckets just have the same values which means it's saying every request is taking more than 180s.

https://prometheus.matrix.org/graph?g0.expr=synapse_room_message_list_rest_servlet_response_time_seconds_bucket&g0.tab=1&g0.stacked=0&g0.show_exemplars=0&g0.range_input=1h&g0.end_input=2022-08-18%2020%3A36%3A02&g0.moment_input=2022-08-18%2020%3A36%3A02
```
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="+Inf", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.005", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.01", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.025", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.05", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.1", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.25", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="0.5", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="1.0", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="10.0", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="120.0", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="180.0", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="2.5", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="30.0", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="5.0", service="synapse"}	79814
synapse_room_message_list_rest_servlet_response_time_seconds_bucket{environment="live", identifier="matrix.org", index="6", instance="matrix.org", job="synapse_client_reader", le="60.0", service="synapse"}	79814
```

Reply:
In any case, we change how this is being timed in https://github.com/matrix-org/synapse/pull/13545/files

Reply:
Fixed in https://github.com/matrix-org/synapse/pull/13584